compiler/semantic/op.go

```diff
@@ -824,3 +831,68 @@ func semCallOp(scope *Scope, call *ast.Call) (dag.Op, error) {
 	}
 	return dag.NewFilter(c), nil
 }
+
+// maybeConvertUserOp returns nil, nil if the call is determined to not be a
+// UserOp, otherwise it returns the compile op or the encountered error.
+func maybeConvertUserOp(ctx context.Context, scope *Scope, call *ast.Call, ds *data.Source, head *lakeparse.Commitish) (dag.Op, error) {
+	op, err := scope.LookupOp(call.Name)
+	if op == nil || err != nil {
+		if _, ok := err.(errOpCycle); !ok {
+			err = nil
+		}
+		return nil, err
+	}
+	if call.Where != nil {
+		return nil, fmt.Errorf("%s(): user defined ops cannot have a where clause", call.Name)
```

Comment:
Nit: s/op/operator/ in all error strings for consistency with the documentation, which uses operator.
```suggestion
		return nil, fmt.Errorf("%s(): user defined operators cannot have a where clause", call.Name)
```